pandas/core/groupby/grouper.py

```diff
@@ -587,20 +588,23 @@ def indices(self):
 
     @property
     def codes(self) -> np.ndarray:
-        if self._passed_categorical:
-            # we make a CategoricalIndex out of the cat grouper
-            # preserving the categories / ordered attributes
-            cat = self.grouper
-            return cat.codes
+        if self._codes is not None:
+            # set in __init__
```

Comment:
Hmm the comment here is a little confusing. So this is set in init but subsequently set to None?

Reply:
self._codes is set in `__init__`, usually to None, but in one case to an ndarray, in which case we use that.  I'll flesh out the comment